biosynseq/metrics.py

```diff
@@ -0,0 +1,529 @@
+
+    Returns
+    -------
+    List[SeqRecord]
+        DNA or protein sequences from the fasta file.
+    """
+    seqs = list(SeqIO.parse(fasta_path, "fasta"))
+    if translate_to_protein:
+        # translate DNA seqs to protein seqs; stop translation at the first in-frame stop codon
+        seqs = [seq.translate(to_stop=True) for seq in seqs]
+    return seqs
+
+
+def fasta_to_dna_seqs(fasta_path: Path) -> List[SeqRecord]:
+    """Given a fasta file, obtain its DNA sequences.
+
+    Parameters
+    ----------
+    fasta_path : Path
+        Path to access the fasta sequences.
+
+    Returns
+    -------
+    List[SeqRecord]
+        DNA sequences from the fasta file.
+    """
+    dna_seqs = list(SeqIO.parse(fasta_path, "fasta"))
+    return dna_seqs
+
+
+def dna_to_protein_seqs(dna_seqs: List[SeqRecord]) -> List[SeqRecord]:
+    """Translate DNA sequenecs to protein sequences.
+    Stop translation at the first in-frame stop codon
+
+    Parameters
+    ----------
+    dna_seqs : List[SeqRecord]
+        List of DNA sequences.
+
+    Returns
+    -------
+    List[SeqRecord]
+        List of protein sequences.
+    """
+    protein_seqs = [seq.translate(to_stop=True) for seq in dna_seqs]
+    return protein_seqs
+
+
+def gc_content(seqs: List[SeqRecord]) -> List:
+    """Given a list of DNA sequences, return each sequence's GC content.
+
+    Parameters
+    ----------
+    seqs : List[SeqRecord.SeqRecord[str]]
+        A list of DNA sequences.
+
+    Returns
+    -------
+    List
+        GC content of each DNA sequence.
+    """
+    return [SeqUtils.GC(rec.seq) for rec in seqs]
+
+
+def seq_length(seqs: List[SeqRecord]) -> List:
+    """Given a list of DNA sequences, return each sequence's length.
+
+    Parameters
+    ----------
+    seqs : List[SeqRecord.SeqRecord[str]]
+        A list of DNA sequences.
+
+    Returns
+    -------
+    List
+        Length of each DNA sequence.
+    """
+    return [len(rec.seq) for rec in seqs]
+
+
+def molecular_weight(protein_seqs: List[SeqRecord]) -> List:
+    """Given a list of protein sequences, return each protein's molecular weight.
+
+    Parameters
+    ----------
+    protein_seqs : List[SeqRecord.SeqRecord[str]]
+        A list of protein sequences.
+
+    Returns
+    -------
+    List
+        Molecular weight of each protein.
+    """
+    return [SeqUtils.molecular_weight(rec.seq, "protein") for rec in protein_seqs]
+
+
+def isoelectric_point(protein_seqs: List[SeqRecord]) -> List:
+    """Given a list of protein sequences, return each protein's isoelectric point.
+
+    Parameters
+    ----------
+    protein_seqs : List[SeqRecord.SeqRecord[str]]
+        A list of protein sequences.
+
+    Returns
+    -------
+    List
+        Isoelectric point of each protein.
+    """
+    return [IsoelectricPoint(seq).pi() for seq in protein_seqs]
+
+
+def compute_alignment_scores(kwargs: dict) -> np.ndarray:
```

Comment:
Have we moved to using v2 functions? If so, this one (and the one that calls it) can be removed.